core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -2798,6 +2917,7 @@ public static JoinRelType simplifyJoin(RelNode joinRel,
    * @param rightFilters list of filters to push to the right child
    * @return whether at least one filter was pushed
    */
+  @Deprecated // to be removed before 2.0
```

Comment:
If we go into this direction, probably the "deprecated flag" it should be added in the javadoc too:
`@deprecated Use newMethod`